candidategenerator.js

```diff
@@ -61,23 +129,29 @@ window.GEN = function(sTree, words, options){
 	var leaves = [];
 	phiNum = wNum = 0;
 	for(var i=0; i<words.length; i++){
-		leaves.push(omegafy(words[i]));
+		leaves.push(wrapInLeafCat(words[i], options.terminalCategory));
 	}
 
 	var recursiveOptions = {};
 	for (var k in options) {
-		if (options.hasOwnProperty(k) && k !== 'requirePhiStem')
+		if (options.hasOwnProperty(k) && k !== 'requireRecWrapper')
 			recursiveOptions[k] = options[k];
 	}
 
-	var rootlessCand = addPhiWrapped(gen(leaves, recursiveOptions), options);
+	/* if rootCategory and recursiveCategory are the same, we don't want to call
+	 * addRecCatWrapped becasue half of the candidates will have a root node with
+	 * only one child, which will be of the same category, ie. {i {i (...) (...)}}
+	 */
+	var rootlessCand = gen(leaves, recursiveOptions)
+	if(options.rootCategory !== options.recursiveCategory)
+	 rootlessCand = addRecCatWrapped(gen(leaves, recursiveOptions), options);
 
 	var candidates = [];
 	for(var i=0; i<rootlessCand.length; i++){
-		var iota = iotafy(rootlessCand[i], options);
+		var iota = wrapInRootCat(rootlessCand[i], options);
```

Comment:
This variable name is still misleading, I'm afraid. Can you change it to proot or root or something else category-neutral?

Reply:
I've changed it to pRoot